OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,79 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean param;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,aliasVarList = {},knownVarList={};
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := getVarsFromEqSystem(inDAE.eqs);
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    if BackendDAEUtil.containAnyVar({cr}, systvars) then  //check in EqSystem.OrderedVars
+      (referencevar,_) := BackendVariable.getVar(cr,systvars);
+    elseif BackendDAEUtil.containAnyVar({cr}, inDAE.shared.globalKnownVars) then //check in shared.globalKnownVars
+      (referencevar,_) := BackendVariable.getVar(cr,inDAE.shared.globalKnownVars);
+    else
+      // some other cases, example variables in shared.externalObjects, and shared.localKnownVars do not handle this case for now
+      referencevar :={};
+    end if;
+    param := checkVariability(referencevar);
+    if (param) then
+      // remove the variable from AliasVarList and add it to GlobalknownVarlist
+      knownVarList := BackendVariable.setVarKind(var,BackendDAE.PARAM()) :: knownVarList;
+      knownVarList := BackendVariable.setVarFixed(var,true) :: knownVarList;
+      //BackendDump.dumpVarList(referencevar,"matchedparamvar");
+    else
+      aliasVarList := var :: aliasVarList;
+    end if;
+    //print("Alias var : " + ComponentReference.crefStr(var.varName) + "==>" + ComponentReference.crefStr(cr) + "\n");
+  end for;
+  //BackendDump.dumpVarList(aliasVarList,"AfterChangingParameters");
+
+  // add the parameter dependent alias vars to Global known Vars after removing from aliasVarList
+  globalKnownVars := BackendVariable.addVars(knownVarList,inDAE.shared.globalKnownVars);
+  outDAE := BackendDAEUtil.setAliasVars(inDAE, BackendVariable.listVar(aliasVarList));
+  outDAE := BackendDAEUtil.setDAEGlobalKnownVars(outDAE, globalKnownVars);
+end fixAliasVarsVariablity;
+
+public function checkVariability "check for variable is PARAM()"
```

Comment:
This function is wrong. Given two variables, one a parameter and one a continuous variable, it would return true instead of false. Is has also a bad name and it should probably be moved to BackendVariable.mo.

Reply:
@lochel  Yes but that is what we discussed and you said if there are 3 variables and if any one of them is a parameter then we should make aliasvariable as parameter

Reply:
No, that's not what I said :D

Reply:
So what should we do in such cases if one variable has parameter and second variable has continuous ?

Reply:
The check should return false in that case.